File: LeetCode_1437/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool kLengthApart(vector<int>& nums, int k) {
        int n = nums.size();
        if (k == 0) return true;
        int last = -1;
        for (int i = 0; i < n; i++) {
            if (nums[i] == 1) {
                if (last == -1) { 
                    last = i;
                } else {
                    if (i - last - 1 < k) {
                        return false;
                    }
                    last = i;
                }
            }
        }
        return true;
    }
};

File: LeetCode_1437/Main.py
from typing import List

class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        if (k == 0):
            return True
        last = -1
        for i in range(n):
            if (nums[i] == 1):
                if (last == -1): 
                    last = i
                else:
                    if (i - last - 1 < k):
                        return False
                    last = i
        return True